### DISCUSSION TITLE
ability to enable/disable motion activated lights on ring floodlights from home assistant

### DISCUSSION THREAD

**Discussion by nomahaffey** [tsightler/ring-mqtt#320] [Feature Requests]
Is it possible to integrate a switch to enable/disable motion activation of any ring flood lights from home assistant? There is a toggle for it in the ring app. It would be helpful to be able to build automations around disabling the motion activated lights for situations where they are more of a nuisance such as when we are just hanging out in the backyard or entertaining guests (the floodlight is constantly going on and off and tends to be fairly blinding).

**Comment by tsightler** [comment]
It would of course be possible to implement this but it would require researching the API commands needed as there's no pre-built function in ring-client-api.  That's not something I have time or interest in doing myself, but if you want to do it and then share that information, I'd be willing to consider implementing it.

You could maybe do some of this with just the standard Ring Modes feature since those can be exposed and controlled by ring-mqtt.  Otherwise you could always use HA for all notificaitons and control the notifications there.

**Reply by nomahaffey** [reply]
Hmm… I’ll poke around and see if I can figure anything out. I haven’t done something like this before so it will be a learning experience even if I don’t end up finding anything…. (I really appreciate your work with  this project btw! Super useful!)

**Reply by tsightler** [reply]
Unfortunately, I don't know much about the camera API, everything that I do know was discovered from looking at ring-client-api or ring-homebridge so if they don't have a given capability it's not clear to me how to implement it.  Sometimes things are a little bit obvious and, if you can perform the action in the Ring web app, you can always use things like Chrome developer tools to see the actual commands, but if it's not possible to do this the web app then you either have to poke at the API manually or use some type of man-in-the-middle proxy to see the HTTP traffic.

For alarm/lighting stuff, using the Chrome tools is the easiest because the web app supports most functions so you can just monitor the websocket connection with Chrome developer tools to see the correct commands, but the camera API is more URL driven and many of the functions are not possible in the web app.

You can check out the python based Ring API which was one of the first to reverse engineer the API using the MITM method and it has a little information on how that was done, although I think Ring has made this a little more difficult in recent years by using pinned certificates in the app.

https://github.com/tchellomello/python-ring-doorbell

I'm guessing you'd need to have some type of system with Magisk to be able to get user installed certificates in the system store so that you can use MITM proxy.  I'd like to have the time to do all this myself, but I just don't right now.

**Reply by broyuken** [reply]
I’m going to vote for this as well. I’d like to make my floodlight camera lights not come on if my deck string lights are on. I don’t have much of a background in doing this sort of stuff either and free time is not plentiful at the moment, but if I somehow have a chance I’ll give it a shot. This would be a super useful addition to ring-mqtt if there is someone with the free time/know how to make it happen before one of us noobs gets a chance to figure it out.

**Comment by broyuken** [comment]
It looks like this ability is not available in the web app. So this is going to be more difficult.

**Reply by tsightler** [reply]
Yep, I had looked at it a few weeks after this thread and didn't see an easy way to do it and just haven't had enough time to try to set up a rooted Android dev system so that I could set up a MITM proxy, not that it would actually take that long, but I just have very little time right now.  I'm hopeful I might have more time in a few months, but that may be optimistic.

**Comment by tsightler** [comment]
Just for future reference:

API endpoint: https://api.ring.com/devices/v1/devices/<doorbotId\>/settings
Method: PATCH

Disable motion activated lights:
```
{
    "enable": true,
    "light_snooze_settings": {
        "always_on": true
    }
}
```

Enable motion activated lights:
```
{
    "enable": false,
    "light_snooze_settings": {
        "always_on": false
    }
}
```

**Comment by broyuken** [comment]
oh sweet! This is different from the motion detection you just added though correct?

**Reply by tsightler** [reply]
Yes, motion detection is different than motion activated lights, no different than in Ring app.

**Comment by broyuken** [comment]
Is this something you’d consider adding? Or should I learn to code and make a PR?

**Reply by tsightler** [reply]
I wouldn't refuse a PR.  :)

But I might get around to adding it at some point, first step was just to get the commands, which I finally managed to setup my Android emulator and mitmproxy so that I could grab.

**Reply by iceman73** [reply]
was wondering on status on this feature add

**Reply by tsightler** [reply]
Status is basically that I barely have time to work on this project at this point, other than the minimum required to keep it working, so new features aren't likely to happen any time soon, unless others step up to contribute.  Maybe that will change at some point this year, but life is just too busy right now.

**Comment by snerd9334** [comment]
+1 on feature request

**Comment by Ev1l-Zer0** [comment]
Also +1 on feature add.

I know it's been some time since this thread has seen any action and as such, I assume there isn't any traction yet. Regardless, is there a positive update to share? 

Thanks,

**Reply by tsightler** [reply]
There are no updates and, realistically, you might be better to ask for such features in the native Home Assistant Ring integration vs here as I plan to remove support for polled devices (Cameras, Chimes, Intercoms) in the next major release for the following reasons:

1) I plan to replace my Ring cameras/doorbell within the next few months, so I will no longer have a way to test them.  
2) The native Ring integration now has 95% of the features for these devices and there aren't really enough non-HA users of these devices to justify the ongoing development effort 
3) For non-HA users that just want camera streams, go2rtc has had native support for adding Ring cameras for some time now, and can be used standalone, so there's really no reason for me to maintain all of the WebRTC stuff in ring-mqtt anymore.

Admittedly, I'm not really working on this project very much these days, so it could be a very long time before any new version is published which would remove these features, probably not until some time next year, but that's the most likely path.

**Reply by broyuken** [reply]
I'll chime in here as well, I just finished ripping out my last ring camera and replaced them with reolink and frigate (so much nicer btw). I want to thank you for all the work you've put in on this addon over the years, but at this point I'm only using it for the alarm, and may be moving off of that as well in the future. I'll have to take a look at the native ring integration to see if that supports the alarm, and if so I may just move over to that instead of running another addon. 

edit: it doesn't support the alarm.

**Reply by tsightler** [reply]
Yes, I'm mostly planning to go with Reolink as well.  Ring is so far behind with camera tech at this point and I don't see anything interesting with them on the horizon.

As far as I know, the native Ring integration still doesn't support Ring Alarm as it uses an entirely different protocol that has never been implemented in the python versions of the Ring API library.  Effectively, this is just the ring-mqtt project returning to it's roots as it was originally called ring-alarm-mqtt and I never had any original intentions of supporting cameras, but users practically begged for live stream support and I was eventually able to get it working, but now that it's in HA natively, very little reason to keep putting that effort into this.

**Comment by omian** [comment]
Since it seems unrealistic that this feature will be added to ring-mqtt, I have added the request to the main integration here: https://github.com/orgs/home-assistant/discussions/1507

If interested, please upvote that request.

**Reply by snerd9334** [reply]
Thank you. I have upvoted it.
